addons/payment_adyen/controllers/main.py

```diff
@@ -120,6 +121,9 @@ def adyen_payments(
             'recurringProcessingModel': 'CardOnFile',  # Most susceptible to trigger a 3DS check
             'shopperIP': payment_utils.get_customer_ip_address(),
             'shopperInteraction': 'Ecommerce',
+            'shopperEmail': partner_sudo.email,
+            'shopperName': partner_sudo.name,
```

Comment:
I'm  not sure how to do it. Should we support different order of first/last names like in OCA?
https://github.com/OCA/partner-contact/blob/e04c0ce733343be788c23da2b2e1af5e38fe8df4/partner_firstname/models/res_partner.py#L170-L184

Reply:
No, I'd just use [`payment`'s util method for splitting names](https://github.com/odoo/odoo/blob/15.0/addons/payment/utils.py#L149). It's far from perfect, but it's consistent with how it's done elsewhere and since quite some years.